odoo/models.py

```diff
@@ -5331,7 +5331,7 @@ def filtered_domain(self, domain):
                         model = model[fname]
                 if comparator in ('like', 'ilike', '=like', '=ilike', 'not ilike', 'not like'):
                     value_esc = value.replace('_', '?').replace('%', '*').replace('[', '?')
-                records = self.browse()
+                records_ids = set()
```

Comment:
It shouldn't be a OrderedSet?  a set doesn't ensure the order and then the filtered_domain won't return the result in the correct order (also not deterministic) ?

Reply:
You are completely right!

Reply:
https://github.com/odoo/odoo/pull/62882

Reply:
@rco-odoo a list will change the behavior compared to the previous code (with union of recordset), right (allow duplicate with list and not in the recordset), I presume that list is more correct ?

Reply:
Note that the union of recorsets preserves "first occurrence" order.

Reply:
But you are right, the `set` discard duplicates, while the `list` does not.